bin-ranges-diff/src/test/java/uk/gov/pay/java_lambdas/bin_ranges_diff/AppTest.java

```diff
@@ -1,17 +1,101 @@
+import static org.mockito.Mockito.atMost;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
 class AppTest {
-    /**
-     * Rigorous Test :-)
-     * Mock s3 etc
-     */
-    @Test
-    void shouldAnswerWithTrue() {
-        assertTrue(true);
+
+    @Mock
+    private Context context;
+
+    @Mock
+    private S3Client mockS3Client;
+    private MockedStatic<Constants> constantsMockedStatic;
+    private MockedStatic<DependencyFactory> dependencyFactoryMockedStatic;
+
+    @BeforeEach
+    public void setUp() {
+        constantsMockedStatic = mockStatic(Constants.class);
+        dependencyFactoryMockedStatic = mockStatic(DependencyFactory.class, CALLS_REAL_METHODS);
+        dependencyFactoryMockedStatic.when(DependencyFactory::s3Client).thenReturn(mockS3Client);
+        when(context.getFunctionName()).thenReturn(this.getClass().getName());
+        when(context.getFunctionVersion()).thenReturn("TEST");
+    }
+
+    @AfterEach
+    public void tearDown() {
+        constantsMockedStatic.close();
+        dependencyFactoryMockedStatic.close();
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void handleRequest_shouldDetermineWhenSHAsAreDifferent(String input, boolean expected) throws IOException {
+        String candidateDataPath = Paths.get("src/test/resources/testData", format("candidate_%s.csv", input)).toString();
+        String promotedDataPath = Paths.get("src/test/resources/testData", format("promoted_%s.csv", input)).toString();
+
+
+        try (FileInputStream candidateFIS = new FileInputStream(candidateDataPath);
+             FileInputStream promotedFIS = new FileInputStream(promotedDataPath)) {
+
+            ResponseInputStream<GetObjectResponse> candidateResponseInputStream = new ResponseInputStream<>(GetObjectResponse.builder().build(), candidateFIS);
+            ResponseInputStream<GetObjectResponse> promotedResponseInputStream = new ResponseInputStream<>(GetObjectResponse.builder().build(), promotedFIS);
+
+            when(mockS3Client.getObject(any(GetObjectRequest.class)))
+                .thenReturn(candidateResponseInputStream)
+                .thenReturn(promotedResponseInputStream);
+
+            App function = new App();
+            Candidate c = new Candidate("/an/s3/key.csv", true);
+            Candidate result = function.handleRequest(c, context);
+            assertEquals(expected, result.proceed());
+
+            verify(mockS3Client, atMost(2)).putObject(any(PutObjectRequest.class), any(RequestBody.class));
+            dependencyFactoryMockedStatic.verify(DependencyFactory::s3Client);
+        }
+    }
+
+    private static Stream<Arguments> handleRequest_shouldDetermineWhenSHAsAreDifferent() {
+        return Stream.of(
+            Arguments.of("same", false),
+            Arguments.of("diff", true)
+        );
```

Comment:
I think using `@CsvSource` directly on the test would be more readable.

Reply:
this involves parsing the boolean from string in the test but I guess that's fine